lib/internal/util.js

```diff
@@ -203,6 +203,41 @@ function getConstructorOf(obj) {
   return null;
 }
 
+function getIdentifiableTagOf(obj) {
+  let namespace = '';
+  let tag = '';
+
+  while (obj) {
+    if (namespace === '') {
+      var desc = Object.getOwnPropertyDescriptor(obj, 'constructor');
+      if (desc !== undefined &&
+         typeof desc.value === 'function' &&
+         desc.value.name !== '')
+        namespace = desc.value.name;
+    }
+
+    if (tag === '') {
+      desc = Object.getOwnPropertyDescriptor(obj, Symbol.toStringTag);
+      if (desc !== undefined) {
+        if (desc.value !== undefined)
+          tag = desc.value;
+        else if (desc.get !== undefined)
+          tag = desc.get();
```

Comment:
`get()` should be called with the `this` object being set to the *original* value of `obj`… right now `this` would refer to the descriptor object

Reply:
Oh, and also maybe wrap this in a `try { … } catch`, we don’t really know what the getter will be doing

Reply:
if it throws why wouldn't we want that to bubble up to the user?